presto-main/src/main/java/com/facebook/presto/operator/JoinOperatorInfo.java

```diff
@@ -103,7 +103,16 @@ public JoinOperatorInfo mergeWith(JoinOperatorInfo other)
             logHistogramProbes[i] = this.logHistogramProbes[i] + other.logHistogramProbes[i];
             logHistogramOutput[i] = this.logHistogramOutput[i] + other.logHistogramOutput[i];
         }
-        return new JoinOperatorInfo(this.joinType, logHistogramProbes, logHistogramOutput, this.lookupSourcePositions + other.lookupSourcePositions);
+
+        long mergedSourcePositions = this.lookupSourcePositions;
+        if (mergedSourcePositions >= 0 && other.lookupSourcePositions >= 0) {
+            mergedSourcePositions += other.lookupSourcePositions;
+        }
+        else {
+            mergedSourcePositions = other.lookupSourcePositions;
```

Comment:
Can it be that `this.lookupSourcePositions > 0` but `other.lookupSourcePositions < 0`?

Reply:
Yes, you're right. Fixing.